### PR TITLE
Add SELinux packages to Mariner.

### PR DESCRIPTION
**Pull request by dburgener**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Enable SELinux support for Mariner

###### Change Log  

This commit add the following packages to Mariner to provide basic SELinux support:
    
    - checkpolicy
    - libsemanage
    - mcstrans
    - policycoreutils
    - secilc
    - selinux-policy
    - setools
    
The selinux-policy provided here is a generic base policy, which is not specifically tuned for Mariner, therefore only permissive mode support is enabled in this commit.  (Although users could load a custom policy to run in enforcing mode).  Future phases have been discussed to add SELinux enforcing mode support.
    
This commit does not enable SELinux by default.  In order to enable SELinux support, one must first install necessary packages (libselinux, policycoreutils, secilc, selinux-policy), and then append "lsm=selinux selinux=1" to the kernel command line.  This will trigger an initial boot to relabel the system, at which point the system will reboot, and boot into an SELinux enabled system.  SELinux state can be queried with the "getenforce" command line tool.  If SELinux has not been enabled, it will report "Disabled" (the default).  If SELinux support has been enabled as described in this paragraph, it will report "permissive".
    
This commit also modifies the following packages to enable SELinux functionality in existing packages:

    - coreutils
    - cronie
    - dbus
    - openssh
    - pam
    - rpm
    - shadow-utils
    - systemd
    - util-list
    
This enables them to build with SELinux support so that when SELinux is enabled, they have SELinux related functionality available.
    
Because coreutils is a basic package and requires building with libselinux-devel present in order to enable key SELinux functionality, several dependencies in other packages that rely on coreutils (namely bash, python2 and python3) had to be removed in order to avoid circular dependencies.  There does not appear to be a functional impact from this change in my testing.

###### Does this affect the toolchain?  

**YES**
NO

###### Associated issues  

- None

###### Links to CVEs  
- None

###### Test Methodology

- Local build and local smoke testing

### CODE REVIEW

File: SPECS/selinux-policy/selinux-policy.spec [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+%pre
+%preInstall refpolicy
+
+%triggerin -- pcre
+selinuxenabled && semodule -nB
+exit 0
+
+%changelog
+* Mon Aug 31 2020 Daniel Burgener <daburgen@microsoft.com> 2.20200818-1
+- Initial creation.  Loosely based on spec from Fedora 31
```
[the remark this answers is not in the document]
Comment (dburgener) [reply SPECS/selinux-policy/selinux-policy.spec]: I switched to the import from Fedora version.  Thanks.

File: SPECS/libsemanage/libsemanage.spec [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+%{_mandir}/man3/*
+
+%files python3
+%{python3_sitelib}/*.so
+%{python3_sitelib}/semanage.py*
+%{_libexecdir}/selinux/semanage_migrate_store
+
+%changelog
+* Tue Aug 25 2020 Daniel Burgener <daburgen@microsoft.com> - 2.9-1
+- Initial import from Fedora 31
```
[the remark this answers is not in the document]
Comment (dburgener) [reply SPECS/libsemanage/libsemanage.spec]: Fix for all, thanks.

File: SPECS/secilc/secilc.spec [hunk 3]
```diff
@@ -0,0 +1,150 @@
+%global libsepolver 2.9-1
+
+Name:           secilc
+Version:        2.9
+Release:        1%{?dist}
```
Comment (PawelWMS) [inline SPECS/secilc/secilc.spec:5]: >1 [](start = 16, length = 1)

This release should actually be 3.

**Comment by PawelWMS** [comment]
> %config %{_sysconfdir}/selinux/refpolicy/contexts/files/media %dir %{_sysconfdir}/selinux/refpolicy/contexts/users

I think the new line got remove here by accident. Or CodeFlow is showing me weird things.

---
Refers to: SPECS/selinux-policy/selinux-policy.spec:72 in 380a4bc. [](commit_id = 380a4bc61fd0d6a907ca9aa6fa39ca2013ded4b3, deletion_comment = False)

**Comment by PawelWMS** [comment]
> * Mon Aug 31 2020 Daniel Burgener <daburgen@microsoft.com> 2.20200818-1

Why the change in versioning format?

---
Refers to: SPECS/selinux-policy/selinux-policy.spec:264 in 380a4bc. [](commit_id = 380a4bc61fd0d6a907ca9aa6fa39ca2013ded4b3, deletion_comment = False)

File: SPECS/setools/setools.spec [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+%{_mandir}/man1/seinfo*
+%{_mandir}/man1/sesearch*
+
+%files python3
+%license COPYING COPYING.GPL COPYING.LGPL
+%{python3_sitearch}/setools
+%{python3_sitearch}/setools-*
+
+%changelog
+* Tue Sep 01 2020 Daniel Burgener <daburgen@microsoft.com> 4.2.2-1
```
Comment (PawelWMS) [inline SPECS/setools/setools.spec]: >1 [](start = 65, length = 1)

Please increment the release.

File: SPECS/setools/setools.spec [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+- Initial CBL-Mariner import from Fedora 31 (license: MIT)
+- License verified
+
+* Mon Jul 08 2019 Vit Mojzis <vmojzis@redhat.com> - 4.2.2-1}
```
Comment (PawelWMS) [inline SPECS/setools/setools.spec]: >} [](start = 59, length = 1)

Was this bracket there in the original? I think we can safely remove it.

File: SPECS/util-linux/util-linux.spec [hunk 6]
```diff
@@ -101,9 +103,10 @@ rm -rf %{buildroot}/lib/systemd/system
 %{_mandir}/man3/*
 
 %changelog
-* Sat May 09 00:20:52 PST 2020 Nick Samson <nisamson@microsoft.com>
-- Added %%license line automatically
-
+*   Fri Sep 04 2020 Daniel Burgener <daburgen@microsoft.com> 2.32.1-4
+-   Enable SELinux support
+*   Sat May 09 2020 Nick Samson <nisamson@microsoft.com>
```
Comment (PawelWMS) [inline SPECS/util-linux/util-linux.spec]: ><nisamson@microsoft.com> [](start = 32, length = 24)

Since you're here: could you add the version info to this line? It seems we're missing "2.32.1-3".

File: SPECS/checkpolicy/checkpolicy.spec [hunk 7]
```diff
@@ -0,0 +1,936 @@
...
+%{_bindir}/checkmodule
+%{_mandir}/man8/checkpolicy.8.gz
+%{_mandir}/man8/checkmodule.8.gz
+%{_mandir}/ru/man8/checkpolicy.8.gz
+%{_mandir}/ru/man8/checkmodule.8.gz
+%{_bindir}/sedismod
+%{_bindir}/sedispol
+
+%changelog
+* Wed Aug 19 2020 Daniel Burgener <Daniel.Burgener@microsoft.com> 2.9-1
```
Comment (PawelWMS) [inline SPECS/checkpolicy/checkpolicy.spec:63]: >1 [](start = 70, length = 1)

The release should be "3".

File: SPECS/mcstrans/mcstrans.spec [hunk 8]
```diff
@@ -0,0 +1,293 @@
+Summary:       SELinux Translation Daemon
+Name:          mcstrans
+Version:       2.9
+Release:       1%{?dist}
+License:       GPLv2+
+Url:           https://github.com/SELinuxProject/selinux/wiki
+Vendor:        Microsoft Corporation
+Distribution:  Mariner
+Source0:       https://github.com/SELinuxProject/selinux/releases/download/20190315/mcstrans-2.9.tar.gz
```
Comment (PawelWMS) [inline SPECS/mcstrans/mcstrans.spec:9]: >mcstrans-2.9.tar.gz [](start = 84, length = 19)

Could you use the `%{name}` and other macros here as well?

**Comment by PawelWMS** [comment]
> Source3:       https://github.com/SELinuxProject/selinux/releases/download/20190315/restorecond-2.9.tar.gz

Please use the `%{version}` macro for these and `%{name}` for the first one.

---
Refers to: SPECS/policycoreutils/policycoreutils.spec:24 in 380a4bc. [](commit_id = 380a4bc61fd0d6a907ca9aa6fa39ca2013ded4b3, deletion_comment = False)

**Review by PawelWMS — CHANGES_REQUESTED** [review]
:clock1:

File: SPECS/bash/bash.spec [hunk 9]
```diff
@@ -259,9 +257,6 @@ make  NON_ROOT_USERNAME=nobody %{?_smp_mflags} check
 
 %post
 if [ $1 -eq 1 ] ; then
-    if [ ! -f "/root/.bash_logout" ] ; then
-        cp /etc/skel/.bash_logout /root/.bash_logout
-    fi
```
[the remark this answers is not in the document]
Comment (lpereira) [reply SPECS/bash/bash.spec:264]: Bootstrapping problems often leads to solutions that aren't elegant -- FWIW, I find that the selinux-headers package idea to be quite alright if it solves the problem

File: SPECS/setools/setools.spec (hunk 5 above)
Comment (dburgener) [reply SPECS/setools/setools.spec]: Yes, that was pulled in from the previous changelog.  I'll remove.

File: SPECS/checkpolicy/checkpolicy.spec (hunk 7 above)
Comment (dburgener) [reply SPECS/checkpolicy/checkpolicy.spec:63]: So in general, we want to continue release numbering from Fedora when doing a spec import?

Comment (PawelWMS) [reply SPECS/checkpolicy/checkpolicy.spec:63]: Yup, pretty much. Unless there's a good reason for a change - doesn't happen to often.

---
In reply to: [486514451](https://github.com/microsoft/CBL-Mariner/pull/100#discussion_r486514451) [](ancestors = 486514451)

**Comment by dburgener** [comment]
> > * Mon Aug 31 2020 Daniel Burgener [daburgen@microsoft.com](mailto:daburgen@microsoft.com) 2.20200818-1
> 
> Why the change in versioning format?
> 
> Refers to: SPECS/selinux-policy/selinux-policy.spec:264 in [380a4bc](https://github.com/microsoft/CBL-Mariner/commit/380a4bc61fd0d6a907ca9aa6fa39ca2013ded4b3). [](commit_id = [380a4bc](https://github.com/microsoft/CBL-Mariner/commit/380a4bc61fd0d6a907ca9aa6fa39ca2013ded4b3), deletion_comment = False)

This package has an interesting history.  Fedora hard forked upstream years ago and their policy has diverged heavily.  We could pull in Fedora's policy, but it will probably make more sense going forward to ship a policy more aligned with upstream.  The versioning previously refers to Fedora's versioning for their fork, and I modified it to match upstream's versioning.

For the current iteration, the details of the policy matter very little because we're not supporting SELinux enforcing mode by default yet.  In permissive mode (what this PR adds support for), SELinux is present and logs what it would deny, but doesn't actually do any access control.  In enforcing mode, it both logs and denies access that is disallowed by the policy.  A customer wanting to take advantage of SELinux at this point would want to supply a policy customized for their use case.  My understanding is that future development plans for Mariner involve creating a Mariner specific SELinux policy that will serve as a suitable generic policy for Mariner users, but that work is not yet scheduled.

In order to actually boot SELinux however, one must have a policy installed, even in permissive mode.  So this commit installs a generic policy which 1. allows us to boot in permissive and 2. provides the basic infrastructure needed for someone to build on top of it for their own policy.  Going with the standard upstream policy rather than Fedora's fork gives us the most flexibility going forward when we decide how to craft Mariner's policy in a future update.  I expect/hope that it will be much more closely tied to upstream, in line with what we are doing with other internal SELinux efforts.  Rather than hard forking, we can add light customizations to upstream to handle our environment appropriately, which will be better for both us and the community.

File: SPECS/secilc/secilc.spec (hunk 3 above)
Comment (dburgener) [reply SPECS/secilc/secilc.spec]: Should be fixed in the latest push.

File: SPECS/setools/setools.spec (hunk 4 above)
Comment (dburgener) [reply SPECS/setools/setools.spec]: Fixed, thanks.

File: SPECS/util-linux/util-linux.spec (hunk 6 above)
Comment (dburgener) [reply SPECS/util-linux/util-linux.spec]: Yep, done.

File: SPECS/checkpolicy/checkpolicy.spec [hunk 10]
```diff
@@ -0,0 +1,936 @@
+%define libselinuxver 2.9-1
+%define libsepolver 2.9-1
+
+Summary:       SELinux policy compiler
+Name:          checkpolicy
+Version:       2.9
+Release:       3%{?dist}
+License:       GPLv2
+Vendor:        Microsoft Corporation
+Distribution:  Mariner
```
Comment (PawelWMS) [inline SPECS/checkpolicy/checkpolicy.spec:10]: >Distribution [](start = 0, length = 12)

I've noticed this just now: this spec is missing the `URL` tag. First time I've seen that btw.:) I'm guessing `[https://github.com/SELinuxProject/selinux`](https://github.com/SELinuxProject/selinux`) is the right one.

File: SPECS/setools/setools.spec [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+rm -rf %{buildroot}/%{_bindir}/sedta
+rm -rf %{buildroot}/%{_bindir}/seinfoflow
+rm -rf %{buildroot}/%{_mandir}/man1/sedta*
+rm -rf %{buildroot}/%{_mandir}/man1/seinfoflow*
+rm -rf %{buildroot}/%{_bindir}/apol
+rm -rf %{buildroot}/%{python3_sitearch}/setoolsgui
+rm -rf %{buildroot}/%{_mandir}/man1/apol*
+
+%files
+%license COPYING COPYING.GPL COPYING.LGPL
```
Comment (PawelWMS) [inline SPECS/setools/setools.spec]: Curious: do you know why the default `setools` is even being built? It looks like it has no runtime dependencies, not pre/post-install steps and it installs no files. Is it possible we can just remove thie one `%file` section?

File: SPECS/systemd/systemd.spec [hunk 12]
```diff
@@ -271,6 +272,8 @@ rm -rf %{buildroot}/*
 %files lang -f %{name}.lang
 
 %changelog
+*  Fri Aug 28 2020 Daniel Burgener <daburgen@microsoft.com> 239-32
+-  Enable SELinux support
```
Comment (PawelWMS) [inline SPECS/systemd/systemd.spec]: >Enable SELinux support [](start = 3, length = 22)

Nit: could you also mention removal of `BuildRequires: shadow-utils` ?

File: SPECS/setools/setools.spec (hunk 11 above)
Comment (dburgener) [reply SPECS/setools/setools.spec]: That was inherited from Fedora.  I'm not sure exactly how it works there.  When I run "dnf deplist setools" on my Fedora box, it lists several dependencies (to other packages in the same spec, such as setools-console), but I don't see them in the Fedora version of the spec.  I don't understand spec files well enough to know how that works.  Does it happen implicitly somehow?

In our context, removing the %file section as you suggest seems fine to me.  I'll make that change.

Comment (PawelWMS) [reply SPECS/setools/setools.spec]: Thank you.

And to answer your question: if there are no explicitly mentioned dependencies in the spec between the default package and one of the subpackages, then I'm also a bit confused, sorry.

---
In reply to: [487020894](https://github.com/microsoft/CBL-Mariner/pull/100#discussion_r487020894) [](ancestors = 487020894)

**Comment by dburgener** [comment]
Is there a tool for updating the cgmanifest files, or is that all supposed to be done by hand?

**Comment by anphel31** [comment]
> Is there a tool for updating the cgmanifest files, or is that all supposed to be done by hand?

File: SPECS/bash/bash.spec (hunk 9 above)
[the remark this answers is not in the document]
Comment (dburgener) [reply SPECS/bash/bash.spec]: I've added the selinux-headers package as discussed here and that seems to address the problem.

File: SPECS/policycoreutils/policycoreutils.spec [hunk 13]
```diff
@@ -0,0 +1,5344 @@
+%global libauditver     3.0
+%global libsepolver     2.9-1
+%global libsemanagever  2.9-1
+%global libselinuxver   2.9-1
+%global sepolgenver     2.9
```
Comment (MateuszMalisz) [inline SPECS/policycoreutils/policycoreutils.spec:5]: This macro appears to be unused.

File: SPECS/policycoreutils/policycoreutils.spec [hunk 14]
```diff
@@ -0,0 +1,5344 @@
+%global libauditver     3.0
+%global libsepolver     2.9-1
```
Comment (MateuszMalisz) [inline SPECS/policycoreutils/policycoreutils.spec:2]: Are these related to this spec's version? If so, I suggest reusing the `%{version}` for their definitions.

File: SPECS/policycoreutils/policycoreutils.spec [hunk 15]
```diff
@@ -0,0 +1,5344 @@
...
+Requires: python3-audit
+Requires: checkpolicy
+Requires: setools-python3 >= 4.1.1
+BuildArch: noarch
+
+%description python3
+The policycoreutils-python3 package contains the interfaces that can be used
+by python 3 in an SELinux environment.
+
+%files python3
```
Comment (MateuszMalisz) [inline SPECS/policycoreutils/policycoreutils.spec:184]: Could you group the `%files` sections together after `%package` and `%description` sections to be consistent with the rest of our specs?

File: SPECS/secilc/secilc.spec [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+%install
+make %{?_smp_mflags} DESTDIR="%{buildroot}" SBINDIR="%{buildroot}%{_sbindir}" LIBDIR="%{buildroot}%{_libdir}" install
+
+
+%files
+%{_bindir}/secilc
+%{_bindir}/secil2conf
+%{_mandir}/man8/secilc.8.gz
+%{_mandir}/man8/secil2conf.8.gz
+%license COPYING
```
Comment (MateuszMalisz) [inline SPECS/secilc/secilc.spec:45]: nit: we usually include `%license` directly after the `%files`.

File: SPECS/secilc/secilc.spec [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+Name:           secilc
+Version:        2.9
+Release:        3%{?dist}
+Summary:        The SELinux CIL Compiler
+License:        BSD
+URL:            https://github.com/SELinuxProject/selinux
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Source0:        %{url}/releases/download/20190315/%{name}-%{version}.tar.gz
+Patch0001:	0001-Allow-setting-arguments-to-xmlto-via-environmental-v.patch
```
Comment (MateuszMalisz) [inline SPECS/secilc/secilc.spec:12]: nit: misaligned fields

File: SPECS/secilc/secilc.spec [hunk 18]
```diff
@@ -0,0 +1,150 @@
+%global libsepolver 2.9-1
```
Comment (MateuszMalisz) [inline SPECS/secilc/secilc.spec:1]: This macro appears to be unused.

File: SPECS/selinux-policy/selinux-policy.spec [hunk 19]
```diff
@@ -0,0 +1,14032 @@
+# upstream does not currently support a mariner distro
+%define distro redhat
```
Comment (MateuszMalisz) [inline SPECS/selinux-policy/selinux-policy.spec]: Could you create a bug for this to be fixed so that we don't lose track of it in the future?

File: SPECS/setools/setools.spec [hunk 20]
```diff
@@ -0,0 +1,182 @@
...
+%package     python3
+Summary:     Policy analysis tools for SELinux
+Obsoletes:   setools-libs < 4.0.0
+Recommends:  libselinux-python3
+Requires:    python3-setuptools
+
+%description python3
+SETools is a collection of graphical tools, command-line tools, and
+Python 3 modules designed to facilitate SELinux policy analysis.
+%prep
```
Comment (MateuszMalisz) [inline SPECS/setools/setools.spec]: nit: please add a newline before `%prep`

File: SPECS/setools/setools.spec [hunk 21]
```diff
@@ -0,0 +1,182 @@
...
+%setup -n %{name}
+
+%build
+pwd
```
Comment (MateuszMalisz) [inline SPECS/setools/setools.spec]: What's the purpose of this `pwd` call?

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh [hunk 22]
```diff
@@ -483,6 +481,47 @@ chroot_and_install_rpms gperf
 chroot_and_install_rpms cryptsetup
 build_rpm_in_chroot_no_install systemd
 
+build_rpm_in_chroot_no_install golang-1.13
+build_rpm_in_chroot_no_install groff
```
Comment (MateuszMalisz) [inline toolkit/scripts/toolchain/build_official_toolchain_rpms.sh]: Are all these packages new additions to the toolchain?

**Review by PawelWMS — APPROVED** [review]
:shipit:

File: SPECS/setools/setools.spec (hunk 21 above)
Comment (dburgener) [reply SPECS/setools/setools.spec]: Was for debugging and I forgot to remove it.  I'll take it out, thanks.

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh (hunk 22 above)
Comment (dburgener) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh]: Yes.  They were needed as dependencies in order to add libsemanage to the toolchain, because shadow-utils picked up a dependency on libsemanage.  I asked @anphel31 about it, and he said that adding in the needed dependencies for libsemanage was the way to go.

File: SPECS/policycoreutils/policycoreutils.spec (hunk 13 above)
Comment (dburgener) [reply SPECS/policycoreutils/policycoreutils.spec]: removed

File: SPECS/policycoreutils/policycoreutils.spec (hunk 14 above)
Comment (dburgener) [reply SPECS/policycoreutils/policycoreutils.spec]: changed

File: SPECS/policycoreutils/policycoreutils.spec (hunk 15 above)
Comment (dburgener) [reply SPECS/policycoreutils/policycoreutils.spec:207]: fixed

File: SPECS/secilc/secilc.spec (hunk 16 above)
Comment (dburgener) [reply SPECS/secilc/secilc.spec:43]: moved

File: SPECS/secilc/secilc.spec (hunk 17 above)
Comment (dburgener) [reply SPECS/secilc/secilc.spec]: fixed

File: SPECS/secilc/secilc.spec (hunk 18 above)
Comment (dburgener) [reply SPECS/secilc/secilc.spec]: removed

File: SPECS/setools/setools.spec (hunk 20 above)
Comment (dburgener) [reply SPECS/setools/setools.spec]: added

File: SPECS/selinux-policy/selinux-policy.spec (hunk 19 above)
Comment (dburgener) [reply SPECS/selinux-policy/selinux-policy.spec]: Sent an e-mail on this.  It's really part of a much larger chunk of work to generally tune the policy to Mariner, rather than just changing things for this one variable.

**Review by anphel31 — APPROVED** [review]

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh (hunk 22 above)
Comment (MateuszMalisz) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh]: Does this mean that these packages should be added to the toolchain manifest as well?

**Review by MateuszMalisz — APPROVED** [review]
If the manifest files don't need updating, LGTM. 🚢

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh (hunk 22 above)
Comment (dburgener) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh]: I don't know the answer to that question, as I don't fully understand the context of what needs to go in the toolchain manifest.  @anphel31 can you help here?  If I added packages into build_official_toolchain_rpms.sh solely in order to provide dependencies for other toolchain packages (not because they themselves are needed as part of the toolchain), do I also need to add them to the toolchain manifests?

File: toolkit/resources/manifests/package/toolchain_aarch64.txt [hunk 23]
```diff
@@ -371,11 +378,13 @@ texinfo-6.5-7.cm1.aarch64.rpm
...
-util-linux-devel-2.32.1-3.cm1.aarch64.rpm
-util-linux-lang-2.32.1-3.cm1.aarch64.rpm
-util-linux-libs-2.32.1-3.cm1.aarch64.rpm
+unzip-6.0-15.cm1.aarch64.rpm
+unzip-debuginfo-6.0-15.cm1.aarch64.rpm
+util-linux-2.32.1-4.cm1.aarch64.rpm
+util-linux-debuginfo-2.32.1-4.cm1.aarch64.rpm
+util-linux-devel-2.32.1-4.cm1.aarch64.rpm
+util-linux-lang-2.32.1-4.cm1.aarch64.rpm
+util-linux-libs-2.32.1-4.cm1.aarch64.rpm
```
Comment (PawelWMS) [inline toolkit/resources/manifests/package/toolchain_aarch64.txt]: It looks like you've removed `util-linux` from being built as part of the toolchain, so we should remove it here as well. Otherwise builds will start failing, since that package will no longer be available in the folder with built toolchain RPMs.

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh (hunk 22 above)
Comment (PawelWMS) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh]: I believe Mateusz is right - we need to mention every package (and its subpackages) we build in the toolchain inside these manifests. It's tedious and annoying - we know. Joe is actually providing automation for that part, so the manifests should go away soon. But probably not before this change gets in.:)

---
In reply to: [498809836](https://github.com/microsoft/CBL-Mariner/pull/100#discussion_r498809836) [](ancestors = 498809836)

Comment (dburgener) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh]: Okay, I'll make the change to add those packages and remove util-linux as mentioned elsewhere.  Once I do that I should probably rerun the toolchain validation pipelines to make sure I didn't make a mistake, so that will take at least 9 hours assuming it's successful on the first run.

Comment (dburgener) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh]: Actually, I don't think util-linux needs removal after all.  I got an e-mail about a comment to that effect, but I can't find it here.  Regardless, it looks to me like util-linux is still in the toolchain, and therefore should remain in the manifests.  I believe that packages in the manifests that weren't built would have been caught by the toolchain pipelines.  It seems as though they don't validate that all built packages are in the manifests though.

Comment (dburgener) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh]: Fixed.  I've kicked off buddy pipeline builds for both toolchains to verify that I didn't make any mistakes in adding them:

https://dev.azure.com/mariner-org/mariner/_build/results?buildId=36188&view=results
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=36189&view=results

**Comment by dburgener** [comment]
I reran the toolchains because of changes to the manifests files and both failed.  I'm looking into the failures now and will restart the toolchains when I have them fixed.

**Comment by dburgener** [comment]
The AMD64 toolchain pipeline completed successfully, but the ARM64 one failed again.  It appears to have been a merge issue, which I have addressed.  I am now rerunning the pipeline.

**Comment by dburgener** [comment]
Both pipelines have completed successfully.  This is ready to be merged from my perspective.
